include/seqan3/std/charconv

```diff
@@ -0,0 +1,1010 @@
+        const uint32_t v1 = static_cast<uint32_t>(value % 100000000);
+        buffer = append4(buffer, v0 / 10000);
+        buffer = append4(buffer, v0 % 10000);
+        buffer = append4(buffer, v1 / 10000);
+        buffer = append4(buffer, v1 % 10000);
+    }
+
+    return buffer;
+}
+
+template <typename _Tp, typename = void>
+struct traits_base
+{
+    using type = uint64_t;
+
+    static int width(_Tp v)
+    {
+        auto t = (64 - __builtin_clzll(v | 1)) * 1233 >> 12;
+        return t - (v < pow10_64[t]) + 1;
+    }
+
+    static char* convert(_Tp v, char* p)
+    {
+        return u64toa(v, p);
+    }
+
+    static auto& pow() { return pow10_64; }
+};
+
+template <typename _Tp>
+struct traits_base<_Tp, decltype(void(uint32_t{declval<_Tp>()}))>
+{
+    using type = uint32_t;
+
+    static int width(_Tp v)
+    {
+        auto t = (32 - __builtin_clz(v | 1)) * 1233 >> 12;
+        return t - (v < pow10_32[t]) + 1;
+    }
+
+    static char* convert(_Tp v, char* p)
+    {
+        return u32toa(v, p);
+    }
+
+    static auto& pow() { return pow10_32; }
+};
+
+template <typename _Tp>
+inline bool mul_overflowed(unsigned char a, _Tp b, unsigned char& r)
+{
+    auto c = a * b;
+    r = c;
+    return c > (numeric_limits<unsigned char>::max)();
+}
+
+template <typename _Tp>
+inline bool mul_overflowed(unsigned short a, _Tp b, unsigned short& r)
+{
+    auto c = a * b;
+    r = c;
+    return c > (numeric_limits<unsigned short>::max)();
+}
+
+template <typename _Tp>
+inline bool mul_overflowed(_Tp a, _Tp b, _Tp& r)
+{
+    static_assert(is_unsigned<_Tp>::value, "");
+    return __builtin_mul_overflow(a, b, &r);
+}
+
+template <typename _Tp, typename _Up>
+inline bool mul_overflowed(_Tp a, _Up b, _Tp& r)
+{
+    return mul_overflowed(a, static_cast<_Tp>(b), r);
+}
+
+template <typename _Tp>
+struct traits : traits_base<_Tp>
+{
+    static constexpr int digits = numeric_limits<_Tp>::digits10 + 1;
+    using traits_base<_Tp>::pow;
+    using typename traits_base<_Tp>::type;
+
+    // precondition: at least one non-zero character available
+    static char const*
+    read(char const* p, char const* ep, type& a, type& b)
+    {
+        type cprod[digits];
+        int j = digits - 1;
+        int i = digits;
+        do
+        {
+            if (!('0' <= *p && *p <= '9'))
+                break;
+            cprod[--i] = *p++ - '0';
+        } while (p != ep && i != 0);
+
+        a = inner_product(cprod + i + 1, cprod + j, pow() + 1,
+                              cprod[i]);
+        if (mul_overflowed(cprod[j], pow()[j - i], b))
+            --p;
+        return p;
+    }
+
+    template <typename _It1, typename _It2, class _Up>
+    static _Up
+    inner_product(_It1 first1, _It1 last1, _It2 first2, _Up init)
+    {
+        for (; first1 < last1; ++first1, ++first2)
+            init = init + *first1 * *first2;
+        return init;
+    }
+};
+
+}  // namespace itoa
+
+template <typename _Tp>
+inline _Tp complement(_Tp x)
```

Comment:
Hmm okay, so `complement` is now public API in std? And the following functions?

Reply:
Hmm... that's what LLVM did but none of the functions is documented in cppreference so I guess they should not appear in the std namespace.
We could add them to the iota namespace? or wrap them in detail?